Entity/TokenManager.php

```diff
@@ -25,27 +25,30 @@ class TokenManager extends BaseTokenManager
      */
     protected $em;
 
-    /**
-     * @var EntityRepository
-     */
-    protected $repository;
-
     /**
      * @var string
      */
     protected $class;
 
     public function __construct(EntityManagerInterface $em, $class)
     {
-        // NOTE: bug in Doctrine, hinting EntityRepository|ObjectRepository when only EntityRepository is expected
-        /** @var EntityRepository $repository */
-        $repository = $em->getRepository($class);
-
         $this->em = $em;
-        $this->repository = $repository;
         $this->class = $class;
     }
 
+    /**
+     * retrocompatibility with old $repository property.
+     * @param $name
+     * @return mixed
+     */
+    public function __get($name)
```

Comment:
should not be public to provide BC with a *protected* property.

and it should trigger a deprecation warning

Reply:
__get can only be public... I can't think a method to provide retrocompatibility for private property.